node/hyperbahn/handler.js

```diff
@@ -337,3 +365,139 @@ function unadvertise(service) {
     var self = this;
     self.channel.topChannel.handler.removeServicePeer(service.serviceName, service.hostPort);
 };
+
+function convertHosts(hosts) {
+    var res = [];
+    for (var i = 0; i < hosts.length; i++) {
+        var strs = hosts[i].split(':');
+        var obj = {
+            port: parseInt(strs[1])
+        };
+        strs = strs[0].split('.');
+        obj.ip = {
+            ipv4: parseInt(strs[3]) + parseInt(strs[2]) << 8 +
+                parseInt(strs[1]) << 16 + parseInt(strs[0]) << 24
+        };
+
+        res.push(obj);
+    }
+
+    return res;
+}
+
+HyperbahnHandler.prototype.discover =
+function discover(self, req, head, body, cb) {
+    var serviceName = body && body.query && body.query.serviceName;
```

Comment:
This statement should be save to have as `var serviceName = body.query.serviceName` again.